pkg/eks/create.go

```diff
@@ -463,3 +474,158 @@ func getParameterValueFromOutput(key string, outputs []*cloudformation.Output) s
 
 	return ""
 }
+
+// EnableEBSCSIDriverInput holds the options for enabling the EBS CSI driver
+type EnableEBSCSIDriverInput struct {
+	EKSService   services.EKSServiceInterface
+	IAMService   services.IAMServiceInterface
+	CFService    services.CloudFormationServiceInterface
+	Config       *eksv1.EKSClusterConfig
+	AddonVersion string
+}
+
+// EnableEBSCSIDriver manages the installation of the EBS CSI driver for EKS, including the
+// creation of the OIDC Provider, the IAM role and the validation and installation of the EKS add-on
+func EnableEBSCSIDriver(opts EnableEBSCSIDriverInput) error {
```

Comment:
```suggestion
func EnableEBSCSIDriver(opts *EnableEBSCSIDriverInput) error {
```
Please, make it a pointer, we changed other functions as well. Sometimes you need to modify, for example, `EKSClusterConfig` that is passed here.